src/main/scala/com/madgag/scalagithub/GitHub.scala

```diff
@@ -220,21 +214,21 @@ class GitHub(ghCredentials: GitHubCredentials) {
     executeAndReadJson(addAuthAndCaching(new Builder().url(repo.trees.urlFor(sha)+"?recursive=1")))
   }
 
-  def followAndEnumerate[T](url: HttpUrl)(implicit ev: Reads[T], ec: EC): Enumerator[Seq[T]] = unfoldM(Option(url)) {
-    _.map { nextUrl =>
+  def followAndEnumerate[T](url: HttpUrl)(implicit ev: Reads[T], ec: EC): Source[Seq[T], NotUsed] = Source.unfoldAsync[Option[HttpUrl],Seq[T]](Some(url)) {
```

Comment:
`play.api.libs.iteratee.Enumerator` is part of the now deceased [`play-iteratees`](https://github.com/playframework/play-iteratees) project, and [`akka.stream.scaladsl.Source`](https://doc.akka.io/api/akka/current/akka/stream/scaladsl/Source.html) is an available candidate for replacing it. Another candidate might have been https://github.com/typelevel/fs2, hopefully I will eventually have time to think/learn about that!

What I'm really after here is some kind of structure that can do a time-consuming thing (doing the IO of hitting an API url) repeatedly, and will eventually know that it's finished by looking at the response of the final IO call. It should operate well with `Future`s (non-blocking) and not make more IO calls that is required by the calling code.